### ISSUE TITLE
Running on with SGE

### ISSUE THREAD

**Issue by ningb** [nf-core/chipseq#118]
Hi,

Thank you for developing this pipeline!

This is not an actual issue, but more of a question. Our institution uses a SGE cluster computing system, with CentOS. Could you please let me know what might be the easiest way to modify and run run your pipeline on such system?

Thanks so much!!

Best Regards

**Comment by drpatelh** [comment]
Hi @ningb . Youre welcome 😃 

I havent personally used SGE but I guess thats the beauty of Nextflow and the way these pipelines are written! You should just be able to create a separate `custom.config` file that has the details for your preferred execution. The `docs` for the Nextflow SGE scope are here:
https://www.nextflow.io/docs/latest/executor.html#sge

For example, I have created one for my Institute because we use Slurm.
https://github.com/nf-core/configs/blob/master/conf/crick.config

You can then just add `-c custom.config` to the command-line and it should work. 

Let me know how you get on.

**Comment by ningb** [comment]
Hi @drpatelh Thanks for your fast response!

I added in the SGE related options to the config file and tried to run it as you suggested. However, one issue I encountered is that I can't run the `--skip_trimming` command (my reads were cleaned and trimmed by the sequencing facility). Nextlow would give me error

`ERROR ~ No signature of method: java.util.ArrayList.ifEmpty() is applicable for argument types: (ArrayList) values: [[]]
Possible solutions: isEmpty(), isEmpty(), first(), first()
 -- Check script 'main.nf' at line: 1469 or see '.nextflow.log' file for more details`

Have you ever experienced this before?

Thanks again!

Best,
Boting

**Comment by drpatelh** [comment]
Hi @ningb . Thats actually a bug in the pipeline 😅 Sorry.

It will be fixed in the next release. These two lines:
https://github.com/nf-core/chipseq/blob/21be3149542cdc84431e12d1e092359058aed32a/main.nf#L499-L500

Should be changed to:
```
    ch_trimgalore_results_mqc = Channel.empty()
    ch_trimgalore_fastqc_reports_mqc = Channel.empty()
```

You should be able to edit those lines in your local version of the pipeline `~/.nextflow/assets/nf-core/chipseq/main.nf`.

Ive submitted a PR for the fix too [#121](https://github.com/nf-core/chipseq/pull/121).

Thanks!

**Comment by drpatelh** [comment]
@ningb This will now be fixed in the next release or you can run the `dev` version of the pipeline by adding `-r dev` to the pipeline command. You may have to delete your local version of the repo first though because it has since been updated.

`rm -r ~/.nextflow/assets/nf-core/chipseq/`

Closing for now but let me know if you have any further problems :+1:
